referenceframe/frame.go

```diff
@@ -445,3 +445,96 @@ func (rf rotationalFrame) MarshalJSON() ([]byte, error) {
 
 	return json.Marshal(temp)
 }
+
+type sevenDFrame struct {
```

Comment:
I don't like calling this a 7D frame even if it takes 7 inputs it really corresponds to a 6D output since there is redundancy in the 4 inputs that comprise the orientation.  Maybe calling it a `poseFrame` or a `mobileFrame` so that the dimensionality is not misleading.